imageio/plugins/spe.py

```diff
@@ -707,26 +708,32 @@ def _get_data(self, index):
             if index >= self._len:
                 raise IndexError("Image index %i > %i" % (index, self._len))
 
+            frame_size = self._shape[0] * self._shape[1]
+            meta = self._get_meta_data(index)
+            try:
+                et = ET.fromstring(meta.get("__xml", ""))
+                (stride,) = {
+                    int(e.attrib["stride"])
+                    for e in et.findall(".//{*}DataBlock[@type='Frame'][@stride]")
```

Comment:
Regarding the x-path:

1. What is the function of the `{*}` here? If I understand correctly, we are actually interested in `DataBlock` elements, so why do we need a wildcard here?
2. Reading the SPE spec (I'm not familiar with intrinsics) I found that `DataBlock`s of type `Frame` always have a `stride` element. Why would we explicitly filter for it here? If a `@type="Frame"` block doesn't have a stride it is malformatted, and (similar to my point below) we would like to `raise` this instead of silently ignoring it.

Reply:
re xpath: the xml is actually namespaced, so in theory you should write something like ".//{http://www.princetoninstruments.com/spe/2009}DataBlock..." which is quite a mouthful for not much semantic information in practice; using a namespace wildcard allows skipping it.

re: stride: same argument as above for lenientness, but again I can change that, just pick either way.

Reply:
re xpath: I see, thanks for explaining

re stride: Yes, I'd prefer a strict implementation. We can always choose to support functionality that goes beyond the  spec itself (like failure recovery in easy cases), but I'd be happier if this happens explicitly. Opt-in or Opt-out both work for me, so we could do something like a `strict=False` kwarg and do error recovery by default. As long as it is possible to fall back to a "_just_ the standard and no-frills" implementation I'm happy :)

Reply:
I just let the errors propagate out for now.